File: atividade_8/java/Livro.java
package java;

public class Livro {
    private String titulo;
    private String autor;
    private int ano;
    private int copias;

    public Livro(String titulo, String autor, int ano, int copias) {
        this.titulo = titulo;
        this.autor = autor;
        this.ano = ano;
        this.copias = copias;
    }

    public void emprestaCopia() {
        this.copias--;
    }

    public void devolveCopia() {
        this.copias++;
    }

    public int verificaCopia() {
        return this.copias;
    }

    @Override
    public String toString() {
        StringBuilder builder = new StringBuilder();
        builder.append("Livro [titulo=").append(titulo).append(", autor=").append(autor).append(", ano=").append(ano)
                .append(", copias=").append(copias).append("]");
        return builder.toString();
    }
}

File: atividade_8/C/livro.c
#include <stdio.h>
#include <string.h>

typedef struct Livro
{
    char titulo[100];
    char autor[30];
    int ano;
    int copias;
};

void inicializar(struct Livro *l, char titulo[], char autor[], int ano, int copias)
{
    strcpy(l->titulo, titulo);
    strcpy(l->autor, autor);
    l->ano = ano;
    l->copias = copias;
}
void emprestaCopia(struct Livro *l)
{
    l->copias--;
}
void devolveCopia(struct Livro *l)
{
    l->copias++;
}
int verificaCopia(struct Livro *l)
{
    return l->copias;
}